src/components/MultiGestureCanvas/usePanGesture.ts

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable no-param-reassign */
+import type {PanGesture} from 'react-native-gesture-handler';
+import {Gesture} from 'react-native-gesture-handler';
+import {useDerivedValue, useSharedValue, withDecay, withSpring} from 'react-native-reanimated';
+import type {CanvasSize, ContentSize, MultiGestureCanvasVariables} from './types';
+import * as MultiGestureCanvasUtils from './utils';
+
+// This value determines how fast the pan animation should phase out
+// We're using a "withDecay" animation to smoothly phase out the pan animation
+// https://docs.swmansion.com/react-native-reanimated/docs/animations/withDecay/
+const PAN_DECAY_DECELARATION = 0.9915;
+
+type UsePanGestureProps = {
+    canvasSize: CanvasSize;
+    contentSize: ContentSize;
+    zoomScale: MultiGestureCanvasVariables['zoomScale'];
+    totalScale: MultiGestureCanvasVariables['totalScale'];
+    offsetX: MultiGestureCanvasVariables['offsetX'];
+    offsetY: MultiGestureCanvasVariables['offsetY'];
+    panTranslateX: MultiGestureCanvasVariables['panTranslateX'];
+    panTranslateY: MultiGestureCanvasVariables['panTranslateY'];
+    isSwipingInPager: MultiGestureCanvasVariables['isSwipingInPager'];
+    stopAnimation: MultiGestureCanvasVariables['stopAnimation'];
```

Comment:
Use union

Reply:
What do you mean by that? This type is used in the hooks to avoid repetition of defining the types

Reply:
Sorry I meant intersection 

```js
type UsePanGestureProps = Pick<MultiGestureCanvasVariables, 'zoomScale' | 'totalScale' | 'offsetX' | 'offsetY' | 'panTranslateX' | 'panTranslateY' | 'isSwipingInPager' | 'stopAnimation'> & {
    canvasSize: CanvasSize;
    contentSize: ContentSize;
};
```